LO-2261: Handle FHIR Datetimes

tdstein opened the pull request:
This change fixes the FHIR date time parsing and migrates from joda-time to the 'java.time' package.

Commit:
Adds date time parsing that supports all scenarios supported in FHIR

Commit:
Replace calls to joda-time with equivalent 'java.time' package (see JSR-310).

Commit:
Removes setting of the system time zone and removes an uncessary print statement

Commit:
Bumps version to 0.4.0.

This is a breaking change to the API and will require users to converts from joda-time to the 'java.time' library

david-wb reviewed (approved):
Nice update. I don't see any problems.

wavetylor commented on src/main/scala/com/lifeomic/fhirlib/v3/Deserializer.scala:
```diff
@@ -1,26 +1,25 @@
+
 import com.lifeomic.fhirlib.v3.datatypes._
 import com.lifeomic.fhirlib.v3.resources.{Resource, _}
-import org.joda.time.format.ISODateTimeFormat
```
If we've removed all joda usages, can we remove it from the POM?

tdstein replied:
Yes - it's already gone. We actually never explicitly imported it. It was grabbing the version from json4s.

Commit:
Converts all dates to UTC during serialization and resolves bug when handling timezones.

By converting each date time to UTC during serialization we simplify the logic for handling dates. The api user always has the option to convert back out of UTC if needed.

mjtieman reviewed (approved):
LGTM, nice work
